PocketKit/Sources/PocketKit/Item/ReadableHostViewController.swift

```diff
@@ -99,7 +91,40 @@ class ReadableHostViewController: UIViewController {
 
     @objc
     private func archiveArticle() {
-        readableViewModel.archiveArticle()
+        readableViewModel.archive()
+    }
+
+    @objc
+    private func unarchiveArticle() {
+        readableViewModel.unarchive { [weak self] success in
+            if success, let archiveButton = self?.getArchiveButton(), let count =  self?.navigationItem.rightBarButtonItems?.count, count >= 3 {
+                self?.navigationItem.rightBarButtonItems?[2] = archiveButton
+            }
```

Comment:
i'd like to see this done a little more safely. if, for whatever reason, index 2 doesn't exist, this will cause a crash. if we do something like, for example, maintain a reference to the unarchive button, we can find it in the right bar button items array (if it exists) and replace it safely.

Reply:
If index 2 doesn't exist, then it shouldn't be able to hit line 101 right? Maintaining a reference will utilize the `lazy var` you suggested right?

Reply:
updated! does this reflect what you were thinking?